2022 Week 44 release (#337)
common-workflow-language/user_guide: src/topics/command-line-tool.md

%
% - Spaces in commands https://github.com/common-workflow-language/user_guide/issues/39
% - Arguments (tell the reader the different use cases for arguments and inputs, tell them there is a section about inputs)

## Network Access

> This indicates whether a process requires outgoing IPv4/IPv6 network access.

Author: Hmm, the placement of this new section seems awkward. None of the other CommandLineTool specific requirements are documented in this section.

I think a new section about network access is fine. We should also explain why network access using steps should be avoided: hurts portability, increases the fragility of the workflow (external service can go away, or change format of responses). We should also present alternatives like putting the URL for an input File in the input object's `location` field instead of a `curl` or `wget` using `CommandLineTool`; and in general separating the automated batch steps from data acquisition and data publishing steps.

Reviewer: Could we add this now and then make a ticket to add this information.  The ticket just said to add this specifically.   https://github.com/common-workflow-language/user_guide/issues/249

> If a command-line tool is written manually in CWL v1.1+, there is a need to 
> specify when network access is required.

Author: ```suggestion
Starting with CWL v1.1, you must document the requirement for network access if it is required.
```

```cwl
cwlVersion: v1.2
class: CommandLineTool

requirements:
  NetworkAccess:
     networkAccess: true
```

```{note}
CWL v1.0 command-line tools that are upgraded to v1.1 

> or v1.2 get Network Access automatically.

Author: ```suggestion
or v1.2 get `networkAccess: true` set automatically.
```
